modules/tinymce/src/core/main/ts/NodeChange.ts

```diff
@@ -49,9 +50,15 @@ class NodeChange {
 
     // IE has a bug where it fires a selectionchange on right click that has a range at the start of the body
     // When the contextmenu event fires the selection is located at the right location
-    editor.on('contextmenu', () => {
-      SelectionBookmark.store(editor);
+    editor.on('contextmenu', (event) => {
+      const elementFromPoint = SugarElement.fromPoint(SugarElement.fromDom(editor.contentDocument), event.x, event.y);
+      elementFromPoint.each((elm) => {
+        if (elm.dom.nodeName === 'A') {
+          editor.selection.select(elm.dom);
```

Comment:
I don't think modifying the selection just because the context menu opened is a good idea.

With a bit of digging it turns out the code that determines whether `open link` should be enabled can deal with selected content that includes a link.
https://github.com/tinymce/tinymce/blob/f6a87ed7a6d32c627cf8ac6c1bd44ce36d69b14b/modules/tinymce/src/plugins/link/main/ts/core/Actions.ts#L83-L87

But the `open link` code itself doesn't use that - it has it's own code to find the link to open, and it's less flexible.
https://github.com/tinymce/tinymce/blob/f6a87ed7a6d32c627cf8ac6c1bd44ce36d69b14b/modules/tinymce/src/plugins/link/main/ts/core/Actions.ts#L9-L13

Just make them both use the more flexible code.

Reply:
Done!